### ISSUE TITLE
issues with luasnip postfix snippets

### ISSUE THREAD

**Issue by hfn92** [Saghen/blink.cmp#769]
### Make sure you have done the following

- [X] I have updated to the latest version of `blink.cmp`
- [X] I have read the README

### Bug Description

I took me a while to start writing this since it seems everyday there is an update that improves this plugin.
So good job and thanks for your effort!

I'm having a few issues to get luasnip to work like I want it to.

I have lots of short snippets that start with special characters such as `,` and `/`. This is mostly to distinguish them
from regular results, so that they don't mix together. I don't want the snippet `ve` to get in conflict with
`std::vector` so that's why I use `,ve` instead.

To get them to trigger I had to modify `completion.keyword.regex` to `[\\[-_,/]\\|\\k`, which I assume is the correct
way to do this?
Though I don't understand the `\\|\\k` part. Is there an explanation that I missed?

But I can't get postfix snippets to trigger consistently. I seems to depend on whether there is a special before the snippet.

Given postfix snippet `\re` for example:

```cpp
Test()/re // transform_items -> get_keyword = /re
```

Will show the completion and expand to `return Test();` correctly.

Wheres

```cpp
Test/re // transform_items -> get_keyword = Test/
```

will only show results from `buffer`.

---

~~The other thing isn't really an issues and more of an idea. But to get `,ve` to have precedence over `std::vector` I had to
set `sources.provides.luasnip.score_offset=3`. This works, but might be to aggressive for other cases. I was thinking
there could be a setting that if a snippet matches exactly, it could get extra score regardless of the fuzzy matching?~~
I guess this can be done manually by using `transform_items` and using `get_keyword` to check all snippets.

### neovim version

NVIM v0.10.2

### `blink.cmp` version: branch, tag, or commit

main

**Comment by hfn92** [comment]
> The other thing isn't really an issues and more of an idea. But to get ,ve to have precedence over std::vector I had to
set sources.provides.luasnip.score_offset=3. This works, but might be to aggressive for other cases. I was thinking
there could be a setting that if a snippet matches exactly, it could get extra score regardless of the fuzzy matching?
I guess this can be done manually by using transform_items and using get_keyword to check all snippets.

Okay so I tried:

```
        luasnip = {
          name = "luasnip",
          transform_items = function(ctx, items)
            local keyword = ctx.get_keyword()
            for _, item in ipairs(items) do
              if keyword == item.label then
                item.score_offset = 50
              end
            end
            return items
          end,
```

But `transform_items` does not get called for each keystroke. Therefore once transformed it will not be called again and I cant update the list. Is there a good way to to this?

**Comment by Saghen** [comment]
> To get them to trigger I had to modify `completion.keyword.regex` to `[\\[-_,/]\\|\\k`, which I assume is the correct
way to do this?

Yeah that's correct, although this is generally hacky since it affects all sources (for example, the path source wouldn't work well with this). #657 will make it so we can have different fuzzy matching ranges per item. I'll adjust the LuaSnip source to return a replace range that includes the `,`. I think that would generally solve this issue so I'll merge this issue into there.

In the mean time, you could try using a binding for showing only the snippet source `keymap['<C-s>'] = { function(cmp) cmp.show({ providers = { 'luasnip' } }) end }`. You could also bind a key to expand the snippets under the cursor, without opening the menu: https://github.com/L3MON4D3/LuaSnip?tab=readme-ov-file#keymaps

> I was thinking there could be a setting that if a snippet matches exactly, it could get extra score regardless of the fuzzy matching?

That should [already be in place](https://github.com/saghen/frizbee?tab=readme-ov-file#algorithm), but I'm guessing the `,` was excluded from the keyword so it wouldn't receive the bonus

> So good job and thanks for your effort!

My pleasure!

**Comment by hfn92** [comment]
@Saghen Thanks! What about the postfix snippets?

**Comment by Saghen** [comment]
Did a bit more digging, and looks like the postfix issue will also be fixed by #657. The postfix isn't matching because including `/` in the `keyword.regex` causes it to match on the whole `Test/re` rather than your desired behavior of just `/re`. I'd recommend that binding that shows only luasnip, combined with the default keyword regex for now
